### PR TITLE
Module 2

### PR DESCRIPTION
**Pull request by waormd**
Module 2 work

### CODE REVIEW

**Comment by Jonesey13** [comment]
Hi @waormd. Could you please upload the PR to Aptem so I can approve/refer after reviewing?

**Comment by waormd** [comment]
Done

File: .gitignore [hunk 1]
```diff
@@ -114,3 +114,6 @@ env/
 
 ### Configuration files ###
 .env
+
+### Secrets ###
+secrets.yml
```
Comment (Jonesey13) [inline .gitignore:119]: Should probably explain how to set this up in the README

File: app.py [hunk 2]
```diff
@@ -1,15 +1,82 @@
 from flask import Flask, render_template, request, redirect, url_for
+from item import Item
 import session_items as session
```
Comment (Jonesey13) [inline app.py:3]: This isn't used anymore

File: app.py [hunk 3]
```diff
@@ -1,15 +1,82 @@
...
+import requests as r
+import yaml
+
+TRELLO_BASE_URL = 'https://api.trello.com'
+TRELLO_BOARD_ID = 'CX81X1uD'
 
 app = Flask(__name__)
 app.config.from_object('flask_config.Config')
```
Comment (Jonesey13) [inline app.py:11]: This line can go (along with the `SECRET_KEY` and `flask_config.py`)

File: app.py [hunk 4]
```diff
@@ -1,15 +1,82 @@
 from flask import Flask, render_template, request, redirect, url_for
+from item import Item
 import session_items as session
+import requests as r
+import yaml
+
+TRELLO_BASE_URL = 'https://api.trello.com'
+TRELLO_BOARD_ID = 'CX81X1uD'
```
Comment (Jonesey13) [inline app.py:8]: This is configuration that probably needs to be extracted from somewhere (and documented in the README)

File: app.py [hunk 5]
```diff
@@ -1,15 +1,82 @@
...
+@app.route('/complete_item', methods=['POST'])
+def indexPut():
+    cardId = request.form['cardId']
+    target = request.form['target']
+    secrets = loadSecrets()
+    lists = loadLists(secrets, TRELLO_BOARD_ID)
+    for item in lists:
+        if item['name'] == target:
+            moveCard(secrets, cardId, item['id'])
+    return index(loadItems(secrets, lists))
```
Comment (Jonesey13) [inline app.py:39]: This doesn't redirect which means the top-level navigation will be in an invalid state (i.e. if I set an item to completed and then attempt to add a ToDo this application will recall this endpoint and fail)

File: app.py [hunk 6]
```diff
@@ -1,15 +1,82 @@
...
+
+def index(sessionItems):
+    return render_template('index.html', items = sessionItems)
```
Comment (Jonesey13) [inline app.py:42]: Unused (and obsolete) default value

File: app.py [hunk 7]
```diff
@@ -1,15 +1,82 @@
...
+
+def loadSecrets():
+    with open('secrets.yml') as file:
+        secrets = yaml.load(file, Loader=yaml.FullLoader)
+        return secrets['trello']
+    return {}
+
+def loadItems(secrets, lists):
+    flattened = []
+    for item in lists:
```
Comment (Jonesey13) [inline app.py:53]: (If you fancy a challenge) You can achieve this with a single list comprehension (you'll need two `for`'s)

File: app.py [hunk 8]
```diff
@@ -1,15 +1,82 @@
...
+        cards = loadCards(secrets, item['id'])
+        for card in cards:
+            flattened.append(Item(card['id'], item['name'], card['name']))
+    return flattened
+
+
+def loadLists(secrets, boardId):
+    url = f'{TRELLO_BASE_URL}/1/boards/{boardId}/lists?key={secrets["api-key"]}&token={secrets["server-token"]}'
```
Comment (Jonesey13) [inline app.py:61]: `requests` has a `params` option which is probably cleaner than using a query string.

File: requirements.txt [hunk 9]
```diff
@@ -1,2 +1,23 @@
+astroid==2.4.2
```
Comment (Jonesey13) [inline requirements.txt:1]: Do we really need all these dependencies?

**Pull request merged**